pkg/observer/observer.go

```diff
@@ -128,7 +128,9 @@ func GetPVC(
 	return result, err
 }
 
-// GetApp fetches the k8s KubeDirectorApp resource in KubeDirector's namespace.
+// GetApp fetches the k8s KubeDirectorApp resource.
+// First check if the app exists in the cluster cr's namespace,
+// if not found check in kubedirector's namespace
 func GetApp(
```

Comment:
So, I'm thinking about the way we use this function during reconcile.

A bunch of the internal catalog functions call GetApp. The first of those calls will invoke observer.GetApp to fetch it (and then cache it).

It seems like that fetch during reconciliation should NOT be doing this namespace-fallback business. During reconciliation we know what the app namespace is, and in fact if we do this code here we might get the wrong app object.

At first glance it looks like:

* The GetApp that we do during validation is good... it should in fact do this code here with the namespace fallbacks.
* The GetApp that we do during reconciliation should be simpler and just try to fetch the app from the known and stored namespace.